exonum/src/runtime/rust/call_context.rs

```diff
@@ -96,6 +97,48 @@ impl<'a> CallContext<'a> {
         Dispatcher::commit_artifact(self.inner.fork, artifact, spec);
     }
 
+    #[doc(hidden)]
+    pub fn initiate_migration(
+        &self,
+        new_artifact: ArtifactId,
+        old_service: &str,
+    ) -> Result<(), ExecutionError> {
+        if self.instance.id != SUPERVISOR_INSTANCE_ID {
+            panic!("`initiate_migration` called within a non-supervisor service");
+        }
+        self.inner
+            .dispatcher
+            .initiate_migration(self.inner.fork, new_artifact, old_service)
+    }
+
+    #[doc(hidden)]
+    pub fn rollback_migration(&self, service_name: &str) -> Result<(), ExecutionError> {
+        if self.instance.id != SUPERVISOR_INSTANCE_ID {
+            panic!("`rollback_migration` called within a non-supervisor service");
+        }
+        Dispatcher::rollback_migration(self.inner.fork, service_name)
+    }
+
+    #[doc(hidden)]
+    pub fn commit_migration(
+        &self,
+        service_name: &str,
+        migration_hash: Hash,
+    ) -> Result<(), ExecutionError> {
+        if self.instance.id != SUPERVISOR_INSTANCE_ID {
```

Comment:
Nit: This can be encapsulated in a method like `ensure_supervisor` / `assert_supervisor`.

Reply:
Access to privileged methods is going to be refactored in #1641, as far as I understand, so I didn't bother with it.